### PR TITLE
Inits Machine Config Daemon

### PR DESCRIPTION
**Pull request by abhinavdahiya**
* adds the basic shim for coordination with MCC.
* sourced from https://github.com/coreos-inc/tectonic-operators/tree/master/node-agent
* does not move test from tectonic-operators
* does not move `updateOS` from https://github.com/coreos-inc/tectonic-operators/blob/fe76a72d4ea3dfd6d9f7177ed751e4d8e379d83e/node-agent/pkg/agent/updater.go#L315-L333

Only implements update and verify for:
Ignition Units, Files

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
#9

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	return nil
+}
+
+// writeFiles writes the given files to disk.
+// it doesn't fetch remote files and expects a flattened config file.
+func (dn *Daemon) writeFiles(files []ignv2_2types.File) error {
+	for _, f := range files {
+		path := dn.prefix + f.Path
+
+		glog.Errorf("Writing file %q", f.Path)
```
Comment (ashcrow) [inline pkg/daemon/update.go]: Should this be `Errorf(..)`?

Comment (abhinavdahiya) [reply pkg/daemon/update.go]: :+1:

File: pkg/daemon/update.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+func (dn *Daemon) reboot() error {
+	glog.Info("Rebooting")
+
+	// reboot
+	dn.loginClient.Reboot(false)
+
+	// cleanup
+	dn.Close()
+
+	// cross fingers
```
Comment (sdemos) [inline pkg/daemon/update.go]: this line will live forever

**Pull request merged**